src/features/island/buildings/components/building/market/Delayed.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { SyntheticEvent, useEffect, useState } from "react";
+import { Button } from "components/ui/Button";
+
+interface Props {
+  restock: (event: SyntheticEvent) => void;
+}
+
+export const Delayed: React.FC<Props> = ({ restock }) => {
+  const [isDisabled, setIsDisabled] = useState(true);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setIsDisabled(false);
+    }, 500);
+    return () => clearTimeout(timer);
+  });
```

Comment:
```suggestion
  }, []);
```

Supplying an empty array as the dependencies will stop this from running more than once.